pkg/suite/suite.go

```diff
@@ -81,3 +94,82 @@ func (s Suite) FindTests(pattern string) ([]runtime.TestCase, error) {
 func (s Suite) GetGlobalConfig() runtime.GlobalTestConfig {
 	return s.Config
 }
+
+// MergeConfigs overlays a global configuration over an entire suite
+// config at the lowest level takes precedence
+func (s Suite) mergeConfigs(config runtime.GlobalTestConfig, nodes []runtime.Node) {
+	s.Config.Env = mergeEnvironmentVariables(s.Config.Env, config.Env)
+
+	if s.Config.Dir == "" {
+		s.Config.Dir = config.Dir
+	}
+
+	if s.Config.Timeout == "" {
+		s.Config.Timeout = config.Timeout
+	}
+
+	if s.Config.Retries == 0 {
+		s.Config.Retries = config.Retries
+	}
+
+	if s.Config.Interval == "" {
+		s.Config.Interval = config.Interval
+	}
+
+	if !s.Config.InheritEnv {
+		s.Config.InheritEnv = config.InheritEnv
+	}
+
+	if len(s.Config.Nodes) == 0 {
+		s.Config.Nodes = config.Nodes
+	}
+
+	// append additional nodes
+	s.Nodes = append(s.Nodes, nodes...)
+
+	s.mergeTestConfigs()
+}
+
+// mergeConfigs will merge the suites runtime.GlobalTestConfig,
+// with each runtime.TestCase in the suite
+func (s Suite) mergeTestConfigs() {
+	for i := range s.TestCases {
```

Comment:
Maybe changing the type of `TestCases` to `[]*runtime.TestCases`, but I am not quite sure why it is not a pointer.

Reply:
I think this change would be better suited for another MR. 
You're talking about changing `suite.Suite` to:

```golang
type Suite struct {
	TestCases []*runtime.TestCase
	Config    runtime.GlobalTestConfig
	Nodes     []runtime.Node
}
```

right? This would require quite a few changes, but I may be misunderstanding.

Reply:
Yes, you are right. Let's stay with this for now